src/akushon/action/model/action.cpp

```diff
@@ -108,6 +107,30 @@ const std::string & Action::get_next_action() const
 void Action::reset()
 {
   poses.clear();
+  joint_splines.clear();
 }
 
+void Action::enable_spline(bool enable)
+{
+  use_spline = enable;
+}
+
+bool Action::is_using_spline() const
+{
+  return use_spline;
+}
+
+void Action::generate_splines()
+{
+  joint_splines.clear();
+  for (auto pose : poses) {
+    for (auto & joint : pose.get_joints()) {
+      uint8_t joint_id = joint.get_id();
+      if (joint_splines.find(joint_id) == joint_splines.end()) {
+        joint_splines[joint_id] = std::make_shared<keisan::SmoothSpline>().get();
+      }
+      joint_splines[joint_id]->add_point(joint.get_position(), pose.get_time());
```

Comment:
Minimizing hash map search:
```suggestion
      const auto it = joint_splines.find(joint.get_id());
      if (it == joint_splines.end()) {
        joint_splines.emplace(joint.get_id(), std::make_shared<keisan::SmoothSpline>().get());
      }
      it.second->add_point(joint.get_position(), pose.get_time());
```